lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/HttpSystemAccessControlModule.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import io.airlift.units.Duration;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+
+public class HttpSystemAccessControlModule
+        implements Module
+{
+    @Override
+    public void configure(Binder binder)
+    {
+        httpClientBinder(binder).bindHttpClient("system-access-control", ForAccessControlRules.class)
+                .withConfigDefaults(config -> config
+                        .setRequestTimeout(Duration.succinctDuration(10, TimeUnit.SECONDS))
+                        .setSelectorCount(1)
+                        .setMinThreads(1));
+        binder.bind(AccessControlRulesRestExtractor.class);
+    }
+
+    @Inject
+    @Provides
+    @Singleton
+    public Supplier<FileBasedSystemAccessControlRules> getFileBasedSystemAccessControlRules(
```

Comment:
Can we have a generic name for this method ? getSystemAccessControlRules

Reply:
done